src/root/views/FinalReportForm/Needs/index.tsx

```diff
@@ -0,0 +1,265 @@
+import React, {
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  PartialForm,
+  Error,
+  EntriesAsList,
+  getErrorObject,
+  useFormArray,
+} from '@togglecorp/toggle-form';
+import {
+  isNotDefined,
+  listToMap,
+  randomString,
+} from '@togglecorp/fujs';
+
+import Container from '#components/Container';
+import languageContext from '#root/languageContext';
+import InputSection from '#components/InputSection';
+import TextArea from '#components/TextArea';
+import NeedInput from '#views/DrefApplicationForm/ActionsFields/NeedInput';
+import Button from '#components/Button';
+import SelectInput from '#components/SelectInput';
+import RadioInput from '#components/RadioInput';
+
+import {
+  booleanOptionKeySelector,
+  BooleanValueOption,
+  DrefFinalReportFields,
+  Need,
+  optionLabelSelector,
+  StringValueOption,
+} from '../common';
+
+import styles from './styles.module.scss';
+
+type Value = PartialForm<DrefFinalReportFields>;
+interface Props {
+  error: Error<Value> | undefined;
+  onValueChange: (...entries: EntriesAsList<Value>) => void;
+  value: Value;
+  yesNoOptions: BooleanValueOption[];
+  needOptions: StringValueOption[];
+}
+
+function Needs(props: Props) {
+  const { strings } = useContext(languageContext);
+
+  const {
+    error: formError,
+    onValueChange,
+    value,
+    yesNoOptions,
+    needOptions,
+  } = props;
+
+  const error = useMemo(() => getErrorObject(formError), [formError]);
+
+  const [need, setNeed] = useState<string | undefined>();
+  const {
+    setValue: onNeedChange,
+    removeValue: onNeedRemove,
+  } = useFormArray<'needs_identified', PartialForm<Need>>(
+    'needs_identified',
+    onValueChange,
+  );
+
+  type Needs = typeof value.needs_identified;
+  const handleNeedAddButtonClick = useCallback((title?: string) => {
+    const clientId = randomString();
+    const newNeedItem: PartialForm<Need> = {
+      clientId,
+      title,
+    };
+
+    onValueChange(
+      (oldValue: PartialForm<Needs>) => (
+        [...(oldValue ?? []), newNeedItem]
+      ),
+      'needs_identified' as const,
+    );
+    setNeed(undefined);
+  }, [onValueChange, setNeed]);
+
+  const needsIdentifiedMap = useMemo(() => (
+    listToMap(
+      value?.needs_identified,
+      need => need.title ?? '',
+      need => true
+    )
+  ), [value.needs_identified]);
+
+  const filteredNeedOptions = useMemo(() => (
+    needsIdentifiedMap ? needOptions.filter(n => !needsIdentifiedMap[n.value]) : []
+  ), [needsIdentifiedMap, needOptions]);
```

Comment:
Isn't this logic inverted?

Reply:
Are we trying to exclude needOptions that are already in needsIdentifiedMap?

Reply:
yes we are excluding needs that are already in needsIdentifiedMap